### DISCUSSION TITLE
Log打印太频繁

### DISCUSSION THREAD

**Discussion by wanglianhui1-tal** [Tencent/libpag#1794] [Q&A] [answered]
[SurfaceTexture-0-29691-7](id:73fb0000000b,api:3,p:29691,c:29691) dequeueBuffer: No free buffer/slot is found!‘’
使用了pagview，如何实现让他停止此log的打印和渲染，并能够重新渲染

**Comment by domchen** [comment, marked as answer]
这个是Android系统打印的，不是PAG。可以去搜索一下系统相关日志的含义。

**Reply by wanglianhui1-tal** [reply]
hello:
我知道这是android 系统打印的，是因为我用了pagview 去play动画的时候，猜测因为它继承TextureView 导致Surface的原因，可目前无论我将此view置为gone，或者add remove view，此log并不会停止打印。打lo发现它的surface并，没有被release

**Reply by domchen** [reply]
你不想让它执行刷新，调用一下stop()就行。把view彻底从显示列表移除不要在添加上去，也会暂停。想立即清理内部的OpenGL环境对Surface的引用，可以叠加一个freeCache的调用。但是调用了这个下次首次刷新时间会变长，从头开始构建环境。

**Reply by wanglianhui1-tal** [reply]
这些方案我这边都测试过。都不能挺住相关log的刷新，因此我采用了pagImageview。但当我导入libpag和使用相关功能时，整体memory增加了70m，这部分如何优化呢

**Reply by wanglianhui1-tal** [reply]
我目前的pagImageview 是放到相关常驻window里的

**Reply by domchen** [reply]
用PAGImageView的话是正解，可以跳过所有视频解码，而且内存现在比实时渲染的PAGView少非常多，你如果觉得70M还是算很多，看看是不是尺寸太大了，调整一下视图的分辨率，PAGImageView的内存占用只跟尺寸有关。

**Comment by tianshaokai** [comment]
@domchen 我调用了stop，有时候日志依然在打印，我使用 PAGView  如何才能停止打印日志呢

**Reply by domchen** [reply]
stop了就不是PAG产生的打印了，找下打印具体内容的归属。而且PAG的打印都会有明确的libpag标识
